src/components/SignUp/index.js

```diff
@@ -1,19 +1,65 @@
-import React from 'react'
+import React, { useState } from 'react'
+import { Link } from 'react-router-dom'
+import { useMutation } from '@apollo/react-hooks'
+import gql from 'graphql-tag'
+
 import styles from './SignUp.module.css'
+import { Form, Button, Jumbotron, Container, Alert } from 'react-bootstrap'
 
-import Form from 'react-bootstrap/Form'
-import Button from 'react-bootstrap/Button'
-import Jumbotron from 'react-bootstrap/Jumbotron'
-import Container from 'react-bootstrap/Container'
-import { Link } from 'react-router-dom'
+const SIGN_UP = gql`
+  mutation sign_up($firstname:String!, $lastname:String!, $username: String!, $password: String!){
+    signup(data:{firstName:$firstname,lastName:$lastname,username:$username,password:$password}){
+      user{
+        id
+        username
+      }
+      jwt
+    }
+  }
+`
 
 function SignUp () {
+  const [show, setShow] = useState(false)
+  const [alertInfo, setAlertInfo] = useState({ variant: 'danger', message: 'Unknown error' })
+
+  const successfulSignup = ({ signup }) => {
+    setAlertInfo({ variant: 'success', message: 'Singed up succesfully' })
+    setShow(true)
+    console.log('success signup', signup)
+  }
+
+  const failureSignup = (error) => {
+    setAlertInfo({ variant: 'danger', message: 'Oh snap! Something went wrong' })
+    setShow(true)
+    console.log(error.message)
+  }
+
+  const [signup] = useMutation(SIGN_UP, {
+    onCompleted: successfulSignup,
+    onError: failureSignup
+  })
+
+  const handleSignup = async (event) => {
+    event.preventDefault()
+    signup({
+      variables: {
+        firstname: event.target.formFirstName.value,
+        lastname: event.target.formLastName.value,
+        username: event.target.formUsername.value,
+        password: event.target.formPassword.value
+      }
+    })
+  }
```

Comment:
Consider using a library such as `Formik` to ease the handling of form state/values/validation.